pcdsdevices/interface.py

```diff
@@ -899,25 +904,29 @@ def set_position(self, position):
         self.set_current_position(position)
 
 
-def setup_preset_paths(**paths):
+def setup_preset_paths(**kwargs):
     """
     Prepare the :class:`Presets` class.
 
     Sets the paths for saving and loading presets.
 
     Parameters
     ----------
-    **paths : str keyword args
+    **kwargs: str keyword args
         A mapping from type of preset to destination path. These will be
         directories that contain the yaml files that define the preset
         positions.
-    """
 
+        (Optional) "defer_loading": bool, whether or not to defer the loading
+        of preset files until the first tab completion
+    """
+    defer_loading = kwargs.pop("defer_loading", False)
+    paths = kwargs
     Presets._paths = {}
     for k, v in paths.items():
         Presets._paths[k] = Path(v)
     for preset in Presets._registry:
-        preset.sync()
+        preset.sync(defer_loading=defer_loading)
```

Comment:
Suggestion: maybe it's simpler to just not call sync at all if `defer_loading` is set to `True`, as long as this doesn't cause any unexpected issues. Then, `sync` doesn't need any code changes.

Reply:
The other half of `sync` sets up the `add_*` methods, which I thought would be nice to have in all cases.  But if we're ok removing those methods then I could get behind it.

Reply:
I forgot about that. I'm not immediately sure how I feel about those methods in the context of having not loaded the presets at all yet.